packages/select/src/components/query-list/queryList.tsx

```diff
@@ -341,6 +398,10 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         }
         Utils.safeInvoke(this.props.onActiveItemChange, activeItem);
     }
+
+    private isCreateItemRendered(): boolean {
+        return !!(this.props.createNewItemFromQuery && this.props.createNewItemRenderer && this.state.query !== "");
```

Comment:
- Use explicit boolean expressions for existence checks.
- Implement `validateProps` to assert (in dev mode at least) that if one of these is provided, both must be provided. Otherwise the create item will just mysteriously never appear.
    - Add unit tests verifying the above.